### PR TITLE
Add date input widget

### PR DESCRIPTION
**Pull request by mrglaser**
This adds a date input widget to forms

ERM 17012

### CODE REVIEW

File: lib/formElement/Date.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	};
+
+	mw.ext.forms.formElement.Date.prototype.getType = function() {
+		return "date";
+	};
+
+	mw.ext.forms.formElement.Date.prototype.getWidgets = function() {
+		return {
+			view: mw.ext.forms.widget.view.TextView,
+			edit: mw.ext.forms.widget.edit.Date
```
Comment (it-spiderman) [inline lib/formElement/Date.js]: If you are not adding anything is the edit widget, no need to have it, you can just use `mw.widgets.DateInputWidget `directly

File: lib/formElement/Date.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+( function ( mw, $, undefined ) {
+	mw.ext.forms.formElement.Date = function() {};
+
+	OO.inheritClass( mw.ext.forms.formElement.Date, mw.ext.forms.formElement.InputFormElement );
+
+	mw.ext.forms.formElement.Date.prototype.getElementConfig = function() {
+		var config = mw.ext.forms.formElement.Date.parent.prototype.getElementConfigInternal.call( this );
+
+		config.value.type = 'date';
```
Comment (it-spiderman) [inline lib/formElement/Date.js]: You could add a control here (for the editor) to specify the `inputFormat` (and/or any other specific configs for this widget)

**Review by it-spiderman — CHANGES_REQUESTED** [review]
Apparently comment here is required

File: lib/formElement/Date.js (hunk 1 above)
Comment (mrglaser) [reply lib/formElement/Date.js]: done

File: lib/formElement/Date.js (hunk 2 above)
Comment (mrglaser) [reply lib/formElement/Date.js]: Actually, there is nothing I need atm. I removed getElementConfig completely

**Pull request merged**